cf/service_access.go

```diff
@@ -140,21 +139,32 @@ func (pc *PlatformClient) UpdateServicePlan(planGUID string, request ServicePlan
 	return servicePlan, nil
 }
 
-func (pc *PlatformClient) getPlanForCatalogPlanGUID(ctx context.Context, catalogPlanGUID string) (cfclient.ServicePlan, error) {
-	plans, err := pc.getServicePlans(ctx, []*types.ServicePlan{&types.ServicePlan{
-		CatalogID: catalogPlanGUID,
-	}})
-
+func (pc *PlatformClient) getPlanForCatalogPlanIDAndBrokerName(ctx context.Context, catalogPlanGUID, brokerName string) (cfclient.ServicePlan, error) {
+	brokers, err := pc.getBrokersByName([]string{brokerName})
 	if err != nil {
 		return cfclient.ServicePlan{}, wrapCFError(err)
 	}
-	if len(plans) == 0 {
-		return cfclient.ServicePlan{}, errors.Errorf("zero plans with catalog plan GUID = %s found", catalogPlanGUID)
+	if len(brokers) == 0 {
+		return cfclient.ServicePlan{}, errors.Errorf("no brokers found for broker name %s", brokerName)
+	}
+	if len(brokers) > 1 {
```

Comment:
not sure if this should be the case. Yes, the sbproxies' broker names are unique but that might not be the case in the future. It's ok for now

Reply:
Should we add some comment here, that we are aware it might break in the future?

Reply:
an issue might be better

Reply:
created